xdsl/irdl/declarative_assembly_format.py

```diff
@@ -433,23 +425,28 @@ def parse(self, parser: Parser, state: ParsingState) -> bool:
                 "assembly format, and thus should not be defined in the attribute "
                 "dictionary."
             )
+
+        parsed = bool(res)
+
+        props = tuple(k for k in res.keys() if k in self.expected_properties)
+        for name in props:
+            state.properties[name] = res.pop(name)
         state.attributes |= res
-        return bool(res)
```

Comment:
seems like a spurious diff

Reply:
I disagree

Reply:
I mean is `parsed` used anywhere other than to return it? Why create a local value then not use it?

Reply:
res is modified after `parsed` is defined

Reply:
Refactored to be nicer

Reply:
Ty